src/mscorlib/shared/System/Memory.cs

```diff
@@ -315,6 +315,16 @@ public unsafe MemoryHandle Retain(bool pin = false)
         /// </summary>
         public bool TryGetArray(out ArraySegment<T> arraySegment)
         {
+            if (_length == 0)
+            {
+#if FEATURE_PORTABLE_SPAN
+                arraySegment = new ArraySegment<T>(SpanHelpers.PerTypeValues<T>.EmptyArray);
+#else
+                arraySegment = ArraySegment<T>.Empty;
+#endif // FEATURE_PORTABLE_SPAN
+                return true;
+            }
```

Comment:
Is doing this check here what we want? Seems like it should instead be done at the end, just before returning false, so that we still call through an OwnedMemory if there is one, and preserve array object identity if this wraps an array, in case that's needed in some situation.

Reply:
Moved to bottom.